Jamroot.jam

```diff
@@ -0,0 +1,20 @@
+path-constant LIBTORRENT_ROOT : . ;
+
+rule asio_src ( properties * )
+{
+	local result ;
+
+	if ! <define>TORRENT_BUILDING_STATIC in $(properties)
```

Comment:
I could check for `<link>static`, but I only want it to be conditional on `! <link>static` for the libtorrent build. Any targets building on top of libtorrent (exe, libtest_common) should include it unconditionally.

I battled with b2 about this for quite a while. What I really wanted to write is,
```
rule asio_src ( properties * )
{
    local result ;
    result += <include>$(LIBTORRENT_ROOT)/include ;
    result += <define>BOOST_ASIO_SEPARATE_COMPILATION ;
    result += <source>$(LIBTORRENT_ROOT)/src/asio.cpp ;
    if <crypto>openssl in $(properties)
    {
        result += <source>$(LIBTORRENT_ROOT)/src/asio_ssl.cpp ;
    }
    return result ;
}
```

and then in the `lib torrent` declaration in the root Jamfile make a requirement like,
```
<link>shared:<conditional>@asio_src
```
However, b2 doesn't seem to handle conditional conditionals properly at all. So I made this workaround instead in order to avoid defining two rules (one for libtorrent with `<link>shared` and one for everyone else like listed above).

An independent project that doesn't build on b2 will just always define their own without worrying about libtorrent defines.